File: problems_001to025/problem_009/solution.py
import os
import sys
from math import ceil

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None

        for a in range(332, 0, -1):
            for b in range(a+1, ceil((1000-a) / 2)):
                c = 1000 - (a + b)

                if a**2 + b**2 == c**2:
                    answer = a * b * c
                    break

            if answer is not None:
                break

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_009/solution.js
const solution = () => {
  let answer = null;

  for (let a = 332; a > 0; a -= 1) {
    for (let b = a + 1; b < Math.ceil((1000 - a) / 2); b += 1) {
      const c = 1000 - (a + b);

      if (a ** 2 + b ** 2 === c ** 2) {
        answer = a * b * c;
        break;
      }
    }

    if (answer !== null) {
      break;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
